chia/rpc/wallet_rpc_api.py

```diff
@@ -984,13 +997,15 @@ async def get_cat_list(self, request) -> EndpointResult:
 
     async def cat_set_name(self, request) -> EndpointResult:
         wallet_id = uint32(request["wallet_id"])
-        wallet: CATWallet = self.service.wallet_state_manager.wallets[wallet_id]
+        wallet = self.service.wallet_state_manager.wallets[wallet_id]
+        assert isinstance(wallet, CATWallet)
```

Comment:
Maybe we want helper functions for this where you pass in the type you want and it returns an instance of that type or raises?  Then we can centralize this check.

Reply:
Not fully thought through, but something of this sort.

```python
T_Wallet = TypeVar("T_Wallet")

def get_wallet_of_type(self, id: int, cls: Type[T_Wallet]) -> T_Wallet:
    wallet = self[id]
    if not isinstance(wallet, cls):
        raise Something()
    return wallet
```

Reply:
that sounds reasonable, at least after exhausted the possibility to move logic into the wallets themselves. (it seems like a code smell to constantly down cast the wallet type)